pulsar-broker/src/main/java/com/yahoo/pulsar/broker/PulsarService.java

```diff
@@ -648,4 +692,36 @@ public String getBrokerServiceUrlTls() {
         return loadManager;
     }
 
-}
+    /**
+     * Looks for a resource in the jar which is expected to be a java.util.Properties, then
+     * extract a specific property value.
+     *
+     * @return the property value, or null if the resource does not exist or the resource
+     *         is not a valid java.util.Properties or the resource does not contain the
+     *         named property
+     */
+    public static String getPropertyFromResource(String resource, String propertyName) {
```

Comment:
Maybe we could move this method and the above `fixVersionString()` into some Utility class?